app/src/main/java/com/sun_asterisk/moviedb_49/data/source/remote/MovieRemoteData.kt

```diff
@@ -0,0 +1,20 @@
+package com.sun_asterisk.moviedb_49.data.source.remote
+
+import com.sun_asterisk.moviedb_49.data.model.Movie
+import com.sun_asterisk.moviedb_49.data.source.MovieDBDataSource
+import com.sun_asterisk.moviedb_49.data.source.remote.fetchdata.GetDataJson
+
+class MovieRemoteData : MovieDBDataSource.Remote {
+
+    override fun getData(listener: OnFetchDataJsonListener<Movie>, string: String) =
+        GetDataJson(listener).getDate(string)
+
+    companion object {
+        private var sInstance: MovieRemoteData? = null
+
+        fun getsInstance(): MovieRemoteData? {
+            if (sInstance == null) sInstance = MovieRemoteData()
+            return sInstance
+        }
```

Comment:
```suggestion
        fun getsInstance(): MovieRemoteData = sInstance ?: MovieRemoteData()       
```
Không thể trả về kiểu Nullable được, có lẽ bạn nên tìm hiểu kĩ hơn về Null Safety trong Kotlin sẽ thú vị hơn 
Trong trường hợp bạn muốn thiết kế một class theo mẫu Singleton, mà không cần truyền tham số nào vào thì bạn có thể sử dụng object trong Kotlin, vì bản thân object trong Kotlin là một class đã được thiết kế singleton design pattern rồi 
```suggestion
        object MovieRemoteData : MovieDBDataSource.Remote {
```
https://antonioleiva.com/objects-kotlin/